nexus/db-queries/src/db/pool_connection.rs

```diff
@@ -47,20 +49,29 @@ impl DieselPgConnector {
         let DieselPgConnectorArgs { user, db, args } = args;
         Self {
             log: log.clone(),
-            prefix: format!("postgresql://{user}@"),
-            suffix: format!(
-                "/{db}{}",
-                args.map(|args| format!("?{args}")).unwrap_or("".to_string())
-            ),
+            user: user.to_string(),
+            db: db.to_string(),
+            args: args
+                .into_iter()
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect(),
         }
     }
 
-    fn to_url(&self, address: std::net::SocketAddr) -> String {
-        format!(
-            "{prefix}{address}{suffix}",
-            prefix = self.prefix,
-            suffix = self.suffix,
-        )
+    fn to_url(
+        &self,
+        address: std::net::SocketAddr,
+    ) -> Result<String, anyhow::Error> {
+        let user = &self.user;
+        let db = &self.db;
+        let mut url =
+            Url::parse(&format!("postgresql://{user}@{address}/{db}"))?;
```

Comment:
i really thought there was a builder-style interface for constructing URLs from parts like this, but looking at the `url` docs, i think it must be the _other_ Rust URL library that has that...